app/livechat/server/business-hour/AbstractBusinessHour.ts

```diff
@@ -0,0 +1,81 @@
+import moment from 'moment';
+
+import { ILivechatBusinessHour, LivechatBussinessHourTypes } from '../../../../definition/ILivechatBusinessHour';
+import {
+	IWorkHoursForCreateCronJobs, LivechatBusinessHoursRaw,
+} from '../../../models/server/raw/LivechatBusinessHours';
+import { UsersRaw } from '../../../models/server/raw/Users';
+import { LivechatBusinessHours, Users } from '../../../models/server/raw';
+
+export interface IBusinessHour {
+	saveBusinessHour(businessHourData: ILivechatBusinessHour): Promise<void>;
+	allowAgentChangeServiceStatus(agentId: string): Promise<boolean>;
+	getBusinessHour(id: string): Promise<ILivechatBusinessHour>;
+	findHoursToCreateJobs(): Promise<IWorkHoursForCreateCronJobs[]>;
+	openBusinessHoursByDayHourAndUTC(day: string, hour: string, utc: string): Promise<void>;
+	closeBusinessHoursByDayAndHour(day: string, hour: string, utc: string): Promise<void>;
+	removeBusinessHoursFromUsers(): Promise<void>;
+	openBusinessHoursIfNeeded(): Promise<void>;
+}
+
+export abstract class AbstractBusinessHour {
+	protected LivechatBusinessHourRepository: LivechatBusinessHoursRaw = LivechatBusinessHours;
+
+	protected UsersRepository: UsersRaw = Users;
+
+	async findHoursToCreateJobs(): Promise<IWorkHoursForCreateCronJobs[]> {
+		return this.LivechatBusinessHourRepository.findHoursToScheduleJobs();
+	}
+
+	async closeBusinessHoursByDayAndHour(day: string, hour: string, utc: string): Promise<void> {
+		const businessHoursIds = await this.LivechatBusinessHourRepository.findActiveBusinessHoursIdsToClose(LivechatBussinessHourTypes.SINGLE, day, hour, utc);
```

Comment:
Why are you hard coding the `LivechatBussinessHourTypes.SINGLE`?
I didn't get it.